app/services/subscribe_service.rb

```diff
@@ -8,28 +8,28 @@ def call(account)
     response     = subscription.subscribe
 
     if response_failed_permanently?(response)
-      # An error in the 4xx range (except for 429, which is rate limiting)
-      # means we're not allowed to subscribe. Fail and move on
+      # We're not allowed to subscribe. Fail and move on.
       account.secret = ''
       account.save!
     elsif response_successful?(response)
-      # Anything in the 2xx range means the subscription will be confirmed
-      # asynchronously, we've done what we needed to do
+      # The subscription will be confirmed asynchronously.
       account.save!
     else
-      # What's left is the 5xx range and 429, which means we need to retry
-      # at a later time. Fail loudly!
+      # The response was either a 429 rate limit, or a 5xx error.
+      # We need to retry at a later time. Fail loudly!
       raise "Subscription attempt failed for #{account.acct} (#{account.hub_url}): HTTP #{response.code}"
     end
   end
 
   private
 
+  # Any response in the 3xx or 4xx range, except for 429 (rate limit)
   def response_failed_permanently?(response)
-    response.code > 299 && response.code < 500 && response.code != 429
+    (response.status.redirect? || response.status.client_error?) && !response.status.too_many_requests?
```

Comment:
Perhaps it's just me but HTTP number codes tell me a lot more than their word names. I know what the 4xx range is, but client_error requires some extra thinking.